src/SmartEntity/FlexEntity.php

```diff
@@ -85,21 +86,64 @@ private function get(string $property)
         return null;
     }
 
-    private function add(string $name, array $arguments)
+    /**
+     * @param string $name
+     * @param array $arguments
+     * @return $this
+     * @throws InvalidArgumentException
+     */
+    protected function add(string $name, array $arguments)
     {
-        if (!isset($this->data[$name]) || is_array($this->data[$name])) {
-            $this->data[$name][] = $arguments[0];
+        if (!isset($this->data[$name])) {
+            $this->data[$name] = [];
+        }
+
+        if (is_array($this->data[$name])) {
+            $this->addArrayField($this->data[$name], $arguments[0], $arguments);
 
             return $this;
         }
 
         throw new InvalidArgumentException('Cannot add to [' . $name . '] property as it is not an array.');
     }
 
-    private function checkArguments(array $arguments, $property): void
+    protected function addArrayField(&$array, $value, array $arguments)
+    {
+        if (isset($arguments[1])) {
+            $array[(string) $arguments[1]] = $value;
+            return $array;
+        }
+
+        $array[] = $value;
+        return $array;
+    }
+
+    protected function fetchValue(array $arguments, $propertyName)
     {
         if (count($arguments) == 0) {
```

Comment:
```suggestion
        if (empty($arguments)) {
```
Empty runs much faster than count, and this function might get called a lot.